### ISSUE TITLE
Cannot enter the process namespace using kubectl alpha debug with target parameter

### ISSUE THREAD

**Issue by ysksuzuki** [kubernetes/kubernetes#98362]
**What happened**:

Cannot enter the process namespace using `kubectl alpha debug` with target parameter.  The nginx process does not appear, but a pause process instead.

```shell
$ kubectl alpha debug -it nginx --image=busybox --target=nginx
Defaulting debug container name to debugger-gs59f.
If you don't see a command prompt, try pressing enter.
/ #
/ # ps ax
PID   USER     TIME  COMMAND
    1 root      0:00 /pause
    7 root      0:00 sh
   13 root      0:00 ps ax
```

**What you expected to happen**:

Enter the process namespace of a target container and see processes running inside it.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a Pod.
```shell
$ cat << EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - name: nginx
    image: nginx
EOF
```
2. Run `kubectl alpha debug` with target parameter and show processes.

```shell
$ kubectl alpha debug -it nginx --image=busybox --target=nginx
Defaulting debug container name to debugger-xgp8k.

If you don't see a command prompt, try pressing enter.

/ #
/ # ps ax
PID   USER     TIME  COMMAND
    1 root      0:00 /pause
    7 root      0:00 sh
   13 root      0:00 ps ax
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.19.7
- OS (e.g: `cat /etc/os-release`):
```
NAME="Flatcar Container Linux by Kinvolk"
ID=flatcar
ID_LIKE=coreos
VERSION=2605.11.0
VERSION_ID=2605.11.0
BUILD_ID=2021-01-08-2026
PRETTY_NAME="Flatcar Container Linux by Kinvolk 2605.11.0 (Oklo)"
ANSI_COLOR="38;5;75"
HOME_URL="https://flatcar-linux.org/"
BUG_REPORT_URL="https://issues.flatcar-linux.org"
FLATCAR_BOARD="amd64-usr"
```
- Kernel (e.g. `uname -a`): `Linux hostname 5.4.87-flatcar #1 SMP Fri Jan 8 19:54:07 -00 2021 x86_64 Intel(R) Xeon(R) Silver 4114 CPU @ 2.20GHz GenuineIntel GNU/Linux`
- Install tools: CKE
- Network plugin and version (if this is a network-related bug): Coil v2.0.5
- Others: Container Runtime: containerd 1.4.3

**Comment by wawa0210** [comment]
Add the `--share-processes` parameter to try

https://kubernetes.io/docs/tasks/debug-application-cluster/debug-running-pod/#debugging-using-a-copy-of-the-pod

**Comment by ysksuzuki** [comment]
I know I can see all processes including nginx and pause with `kubectl alpha debug nginx -it --image=ubuntu --share-processes --copy-to=myapp-debug`, but why `--target` parameter doesn't work as expected on my environment? I thought it targets the process namespace of another container.

**Comment by eddiezane** [comment]
/assign @verb 

Please take a look and accept if so.

**Comment by verb** [comment]
I suspect this is because containerd hasn't implemented [NamespaceMode_TARGET](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/495-pod-pid-namespace#targeting-a-specific-containers-namespace)

This works for me with the builtin dockershim:

```
% kc version
Client Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.7-dirty", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"dirty", BuildDate:"2021-02-04T17:13:43Z", GoVersion:"go1.15.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.7-dirty", GitCommit:"1dd5338295409edcfff11505e7bb246f0d325d15", GitTreeState:"dirty", BuildDate:"2021-02-04T17:13:43Z", GoVersion:"go1.15.3", Compiler:"gc", Platform:"linux/amd64"}
 --- 2021-02-04 18:32 ---

% cat << EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - name: nginx
    image: nginx
EOF

pod/nginx created
 --- 2021-02-04 18:33 ---

% kubectl alpha debug -it nginx --image=busybox --target=nginx
Defaulting debug container name to debugger-nkpp8.
If you don't see a command prompt, try pressing enter.
/ # ps ax
PID   USER     TIME  COMMAND
    1 root      0:00 nginx: master process nginx -g daemon off;
   28 101       0:00 nginx: worker process
   29 root      0:00 sh
   34 root      0:00 ps ax
/ #
```

I know that dockershim is slated for removal soon, so we should ask containerd to support this.

**Comment by verb** [comment]
We should probably print a warning that `--target` requires support from the runtime

**Comment by verb** [comment]
Opened https://github.com/containerd/containerd/issues/5008 to discuss this with the containerd authors.

**Comment by akifkhan01** [comment]
> I know I can see all processes including nginx and pause with `kubectl alpha debug nginx -it --image=ubuntu --share-processes --copy-to=myapp-debug`, but why `--target` parameter doesn't work as expected on my environment? I thought it targets the process namespace of another container.

@ysksuzuki Did you try this with `--share-processes` ?

I tried which didn't work for me with the message :
Error: unknown flag: --share-processes
See 'kubectl alpha debug --help' for usage. ( And this doesn't list --share-processes as an option )

this suggests me that kubectl alpha debug doesn't have --share-processes option enabled at least until kubectl version v1.18.4

Environment :  
Kubernetes version (use kubectl version): 1.18.4
Others: Container Runtime: containerd 1.3.3

**Comment by verb** [comment]
containerd now has an open PR to fix this (containerd/containerd#5203)

**Comment by verb** [comment]
containerd/containerd#5008 has been resolved. containerd plans to support namespace targeting in the 1.5 release.
